### PR TITLE
add aodh service role

### PR DESCRIPTION
**Pull request by zfeldstein**

### CODE REVIEW

File: roles/aodh/templates/etc/aodh/policy.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (nirajdp76) [inline roles/aodh/templates/etc/aodh/policy.json]: will need adjustment for cloud_admin and project_admin

File: roles/endpoints/defaults/main.yml [hunk 2]
```diff
@@ -226,3 +226,18 @@ endpoints:
...
+    name: aodh
+    type: alarming
+    description: 'Alarming Service'
+    url:
+      admin: https://{{ fqdn }}:8042
+      internal: https://{{ fqdn }}:8042
+      public: https://{{ fqdn }}:8042
+    port:
+      backend_api: 8043
+      haproxy_api: 8042
```
Comment (nirajdp76) [inline roles/endpoints/defaults/main.yml]: Will need to get this front-end port (8042)added to our firewall rules (vyatta).

File: roles/aodh/templates/etc/aodh/policy.json (hunk 1 above)
Comment (zfeldstein) [inline roles/aodh/templates/etc/aodh/policy.json]: Do we want to put cloud_admin and project_admin in the context_is_admin rule?

File: roles/aodh/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
+---
+- name: Check if aodh user exists
+  action: shell getent passwd aodh
+  register: aodh_user
+  failed_when: False
+  changed_when: False
+
+- name: aodh user
+  user: name=aodh shell=/bin/false createhome=no
+  when: aodh_user|success
```
Comment (j2sol) [inline roles/aodh/tasks/main.yml]: The user module is idempotent, why are we checking first and adding after? Why not just use the user module?

**Comment by zfeldstein** [comment]
retest

File: roles/aodh/templates/etc/aodh/aodh.conf [hunk 4]
```diff
@@ -0,0 +1,52 @@
+[DEFAULT]
+auth_strategy = keystone
+rpc_backend = rabbit
+debug = {{ aodh.debug }}
+
+log_dir = /var/log/aodh
+debug = {{ aodh.debug }}
```
Comment (j2sol) [inline roles/aodh/templates/etc/aodh/aodh.conf]: Duplicate `debug` lines.

File: roles/aodh/templates/etc/aodh/policy.json (hunk 1 above)
Comment (j2sol) [inline roles/aodh/templates/etc/aodh/policy.json]: Looks right, I can't see anything here that would be dangerous for cloud_admin to do.

File: roles/aodh/templates/etc/serverspec/aodh_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		     '}'].join("\n")
+
+  it { should contain file_contents }
+end
+
+describe file('/etc/aodh/aodh.conf') do
+  it { should contain 'debug = {{ aodh.debug }}' }
+  it { should contain 'log_dir = /var/log/aodh' }
+  it { should contain 'os_auth_url = {{ endpoints.keystone.url.admin }}' }
+end
```
Comment (j2sol) [inline roles/aodh/templates/etc/serverspec/aodh_spec.rb]: Do we have tech spec entries for all these things we're looking for? I think we're supposed to reference line items somehow.

File: roles/endpoints/defaults/main.yml [hunk 6]
```diff
@@ -226,3 +226,18 @@ endpoints:
...
+    haproxy:
+      health_check: 'tcp-check /'
```
Comment (j2sol) [inline roles/endpoints/defaults/main.yml]: Can we use the health check middleware for this, like we did in glance?

**Comment by j2sol** [comment]
Overall looks solid.  Just a few nits and questions.

**Pull request merged**